src/services/completions.ts

```diff
@@ -5084,17 +5108,22 @@ function isModuleSpecifierMissingOrEmpty(specifier: ModuleReference | Expression
     return !tryCast(isExternalModuleReference(specifier) ? specifier.expression : specifier, isStringLiteralLike)?.text;
 }
 
-function getVariableDeclaration(property: Node): VariableDeclaration | undefined {
-    const variableDeclaration = findAncestor(property, node =>
-        isFunctionBlock(node) || isArrowFunctionBody(node) || isBindingPattern(node)
-            ? "quit"
-            : isVariableDeclaration(node));
+function getVariableOrParameterDeclaration(contextToken: Node | undefined) {
+    if (!contextToken) return;
 
-    return variableDeclaration as VariableDeclaration | undefined;
+    return findAncestor(contextToken, node =>
+        ((isParameter(node) || isTypeParameterDeclaration(node)) && !isIndexSignatureDeclaration(node.parent)) ||
+            (isFunctionBlock(node) || isArrowFunctionBody(node) || isBindingPattern(node)
+            ? "quit"
+            : isVariableDeclaration(node))) as ParameterDeclaration | TypeParameterDeclaration | VariableDeclaration | undefined;

```

Comment:
1. The `"quit"` check should be first.
2. When the parameter checks are moved down with the variable check, the line becomes long enough that the original two-line temp+cast becomes easier to read again.

```suggestion
    const declaration = findAncestor(contextToken, node =>
        isFunctionBlock(node) || isArrowFunctionBody(node) || isBindingPattern(node)
            ? "quit"
            :  isVariableDeclaration(node) || ((isParameter(node) || isTypeParameterDeclaration(node)) && !isIndexSignatureDeclaration(node.parent)));
    return declaration as ParameterDeclaration | TypeParameterDeclaration | VariableDeclaration | undefined;
```